src/js/components/DataTable/Header.js

```diff
@@ -457,25 +446,19 @@ const Header = forwardRef(
                     style={onResize ? { position: 'relative' } : undefined}
                   >
                     {content}
-                    {searcher && resizer ? (
-                      <Box
-                        flex="shrink"
-                        direction="row"
-                        align="center"
-                        gap={theme.dataTable.header.gap}
-                      >
-                        {searcher}
-                        {resizer}
-                      </Box>
+                    {searcher && onResize ? (
+                      <Box pad={{ right: 'xsmall' }}>{searcher}</Box>
                     ) : (
-                      searcher || resizer
+                      searcher
                     )}
                   </Box>
                 );
               }
               const cellPin = [...pin];
               if (columnPin) cellPin.push('left');
 
+              const headerId = `grommet-data-table-header-${property}`;
```

Comment:
does this help with the accessibility I noticed we added the `id` below with the name?

Reply:
Added back role seperator (see comment here: https://github.com/grommet/grommet/pull/7716#discussion_r2220453783)